From 3f63a4b3264cd07be766be6e42b7bbce24a14fc8 Mon Sep 17 00:00:00 2001
From: Dave Eddy <dave@daveeddy.com>
Date: Thu, 29 Mar 2018 14:05:32 -0400
Subject: [PATCH] TRITON-280 vmapi vms.full.test should block on test VMs to be
 destroyed fully Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
 Approved by: Todd Whiteman <todd.whiteman@joyent.com>

---
 test/vms.full.test.js | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/test/vms.full.test.js b/test/vms.full.test.js
index a4534dc..a292f0f 100644
--- a/test/vms.full.test.js
+++ b/test/vms.full.test.js
@@ -184,6 +184,9 @@ function createTestVms(cb) {
                     return;
                 }
 
+                assert.object(body, 'body');
+                assert.uuid(body.job_uuid, 'body.job_uuid');
+
                 var job = '/jobs/' + body.job_uuid;
 
                 i++;
@@ -241,8 +244,22 @@ function destroyTestVms(cb) {
                         return;
                     }
 
-                    ret.destroyedVms++;
-                    next();
+                    assert.object(delBody, 'delBody');
+                    assert.uuid(delBody.job_uuid, 'delBody.job_uuid');
+
+                    var job = '/jobs/' + delBody.job_uuid;
+
+                    waitForValue(job, 'execution', 'succeeded', {
+                        client: client
+                    }, function (waitForValueErr) {
+                        if (waitForValueErr) {
+                            next(waitForValueErr);
+                            return;
+                        }
+
+                        ret.destroyedVms++;
+                        next();
+                    });
                 });
 
             }, function (delVmsErr) {
-- 
2.21.0

